packages/components/src/autocomplete/index.tsx

```diff
@@ -146,11 +90,20 @@ function useAutocomplete( {
 		if ( getOptionCompletion ) {
 			const completion = getOptionCompletion( option.value, filterValue );
 
-			const { action, value } =
-				undefined === completion.action ||
-				undefined === completion.value
-					? { action: 'insert-at-caret', value: completion }
-					: completion;
+			function isOptionCompletion(
+				completionData: any
+			): completionData is OptionCompletion {
+				return (
+					undefined === completionData.action ||
+					undefined === completionData.value
+				);
+			}
+			const action = isOptionCompletion( completion )
+				? completion.action
+				: 'insert-at-caret';
+			const value = isOptionCompletion( completion )
+				? completion.value
+				: completion;
```

Comment:
Runtime change: custom typeguard and new ternaries. The original ternary wasn't able to adequately narrow `completion` to avoid errors (unless it can, in which case I'd love to learn that 🪄 )